commit 62668a22194bcf816384a4ffd003ee3b52d592fa (refs/changes/28/3228/2)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2018-01-17T21:03:18+00:00 (1 year, 9 months ago)
    
    VOLAPI-101 docker test assertion failure when NFS support is disabled
    Reviewed by: Dave Eddy <dave.eddy@joyent.com>
    Approved by: Dave Eddy <dave.eddy@joyent.com>

diff --git a/test/integration/api-create-invalid-names.test.js b/test/integration/api-create-invalid-names.test.js
index 1dbe23f..7b00708 100644
--- a/test/integration/api-create-invalid-names.test.js
+++ b/test/integration/api-create-invalid-names.test.js
@@ -8,9 +8,9 @@
  * Copyright 2018, Joyent, Inc.
  */
 
-var test = require('tape');
 var vasync = require('vasync');
 
+var mod_testVolumes = require('../lib/volumes');
 var testHelpers = require('./helpers');
 var volumesApi = require('../lib/volumes-api');
 
@@ -20,6 +20,10 @@ var STATE = {
     log: require('../lib/log')
 };
 
+var test = mod_testVolumes.createTestFunc({
+    checkTritonSupportsNfsVols: true
+});
+
 test('setup', function (tt) {
 
     tt.test('docker env', function (t) {
@@ -52,7 +56,7 @@ test('api: create volumes with invalid name', function (tt) {
          * 'x'.repeat(257) generates a volume name that is one character too
          * long, as the max length for volume names is 256 characters.
          */
-        var INVALID_VOLUME_NAMES = ['', '-foo', '.foo', 'x'.repeat(257)];
+        var INVALID_VOLUME_NAMES = ['-foo', '.foo', 'x'.repeat(257)];
 
         vasync.forEachParallel({
             func: function createVolume(volumeName, done) {
diff --git a/test/lib/volumes-api.js b/test/lib/volumes-api.js
index 94083e7..a74516c 100644
--- a/test/lib/volumes-api.js
+++ b/test/lib/volumes-api.js
@@ -39,8 +39,6 @@ function createDockerVolume(opts, callback) {
     assert.optionalString(opts.name, 'opts.name');
     assert.func(callback, 'callback');
 
-    assert.ok(testVolumes.nfsSharedVolumesSupported());
-
     var dockerClient = opts.dockerClient;
     var dockerApiVersion = opts.apiVersion || ('v' + constants.API_VERSION);
 
